drivers/src/main/java/org/gorpipe/oci/driver/OCIUrl.java

```diff
@@ -107,10 +109,6 @@ private static OCIUrl parseHttpUrl(URI uri) throws MalformedURLException {
 
     //  {@literal https://<oci-endpoint>/n/<namespace>/b/<bucket>/o/<path>/<to>/<file>}
     private static OCIUrl parseOCIHttpUrl(URI uri) throws MalformedURLException {
-        if (!Arrays.asList("http", "https", "s3").contains(uri.getScheme())) {
-            throw new MalformedURLException("S3 url must have http[s]/s3 scheme: " + uri);
-        }
-
         var result = new OCIUrl();
 
         result.endpoint = uri.getAuthority();
```

Comment:
A lot of hardcoded indices, are we sure that this is always the same?

Reply:
Which one are you talking about.  The structure of the urls follows a pattern, so many of them should be fixed.